pkg/tagset/default.go

```diff
@@ -0,0 +1,46 @@
+package tagset
+
+// DefaultFactory is a global thread-safe factory, used by calls to
+// package-level functions. This is suitable for non-performance-critical tags
+// manipulation
+var DefaultFactory Factory
+
+// EmptyTags is a ready-to-use Tags instance that contains no tags. Use this
+// instead of nil values for *Tags.
+var EmptyTags *Tags
+
+func init() {
+	cf, _ := NewCachingFactory(1000, 5)
```

Comment:
Should 1000 be a tunable config value?

Reply:
Per discussion today, I replaced this with a non-caching factory.  Eventually, yes, it may be (or perhaps will tune itself).